actors/builtin/miner/miner_actor.go

```diff
@@ -991,49 +991,54 @@ func (a Actor) commitWorkerKeyChange(rt Runtime) *adt.EmptyValue {
-		rt.Abortf(exitcode.ErrIllegalArgument, "owner must be an ID address: %v", address)
+// Resolves an address to an ID address and verifies that it is address of an account or multisig actor.
+func resolveOwnerAddress(rt Runtime, raw addr.Address) addr.Address {
+	resolved, ok := rt.ResolveAddress(raw)
+	if !ok {
+		rt.Abortf(exitcode.ErrIllegalArgument, "unable to resolve address %v", raw)
 	}
-	ownerCode, ok := rt.GetActorCodeCID(address)
+	Assert(resolved.Protocol() == addr.ID)
+
+	ownerCode, ok := rt.GetActorCodeCID(resolved)
 	if !ok {
-		rt.Abortf(exitcode.ErrIllegalArgument, "no code for address %v", address)
+		rt.Abortf(exitcode.ErrIllegalArgument, "no code for address %v", resolved)
 	}
 	if !builtin.IsPrincipal(ownerCode) {
 		rt.Abortf(exitcode.ErrIllegalArgument, "owner actor type must be a principal, was %v", ownerCode)
 	}
+	return resolved
 }
 
-// Verifies that an address is the ID address of an account with an associated BLS key.
+// Resolves an address to an ID address and verifies that it is address of an account actor with an associated BLS key.
 // The worker must be BLS since the worker key will be used alongside a BLS-VRF.
-func verifyAddressAsWorker(rt Runtime, address addr.Address) {
-	// Validate the worker address is an ID address of an account actor with a BLS key.
-	if address.Protocol() != addr.ID {
-		// It would be convenient to be able to accept a BLS address here, which would be safe to use without waiting
-		// for finality of a newly-created account actor ID-address.
-		// This would require the runtime to provide a pubkey -> ID address resolution mechanism so we can persist
-		// the ID address in state.
-		rt.Abortf(exitcode.ErrIllegalArgument, "worker must be an ID address: %v", address)
+func resolveWorkerAddress(rt Runtime, raw addr.Address) addr.Address {
+	resolved, ok := rt.ResolveAddress(raw)
+	if !ok {
+		rt.Abortf(exitcode.ErrIllegalArgument, "unable to resolve address %v", raw)
 	}
-	ownerCode, ok := rt.GetActorCodeCID(address)
+	Assert(resolved.Protocol() == addr.ID)
+
+	ownerCode, ok := rt.GetActorCodeCID(resolved)
 	if !ok {
-		rt.Abortf(exitcode.ErrIllegalArgument, "no code for address %v", address)
+		rt.Abortf(exitcode.ErrIllegalArgument, "no code for address %v", resolved)
 	}
 	if ownerCode != builtin.AccountActorCodeID {
 		rt.Abortf(exitcode.ErrIllegalArgument, "worker actor type must be an account, was %v", ownerCode)
 	}
 
-	ret, code := rt.Send(address, builtin.MethodsAccount.PubkeyAddress, &adt.EmptyValue{}, big.Zero())
-	builtin.RequireSuccess(rt, code, "failed to fetch account pubkey from %v", address)
-	var pubkey addr.Address
-	err := ret.Into(&pubkey)
-	if err != nil {
-		rt.Abortf(exitcode.ErrSerialization, "failed to deserialize address result: %v", ret)
-	}
-	if pubkey.Protocol() != addr.BLS {
-		rt.Abortf(exitcode.ErrIllegalArgument, "worker account %v must have BLS pubkey, was %v", address, pubkey.Protocol())
+	if raw.Protocol() != addr.BLS {
```

Comment:
Leave the BLS check out until go-filecoin has BLS fixed. 
We can't build the genesis block with this check in.

Reply:
It looks like the reported non-working of BLS in go-filecoin may be based on a misunderstanding. I'm going to land this in the desired form, and if more BLS problems arise that'll be my top priority.